gemfirexd/core/src/main/javacc/com/pivotal/gemfirexd/internal/impl/sql/compile/sqlgrammar.jj

```diff
@@ -5496,6 +5510,10 @@ longSubType() throws StandardException :
 		throw StandardException.newException(SQLState.NOT_IMPLEMENTED, TypeId.NATIONAL_LONGVARCHAR_NAME);
 		// return DataTypeDescriptor.getBuiltInDataTypeDescriptor(TypeId.NATIONAL_LONGVARCHAR_NAME);
 	}
+|
+    {
+        return DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.BIGINT);
+    }
```

Comment:
That's good option and avoids LOOKAHEAD. Confirm that store precheckin works fine and grammar does not lead to incorrect code due to ambiguity.